internal/controller/controllers/preprovisioningimage_controller.go

```diff
@@ -428,25 +428,34 @@ func (r *PreprovisioningImageReconciler) getIronicAgentImageFromUserOverride(log
 	return ""
 }
 
-func (r *PreprovisioningImageReconciler) getIronicConfigFromBMOConfig(ctx context.Context, log logrus.FieldLogger, infraEnvInternal *common.InfraEnv) *ICCConfig {
+func (r *PreprovisioningImageReconciler) getIronicConfigFromBMOConfig(ctx context.Context, log logrus.FieldLogger) *ICCConfig {
 	iccConfig, err := r.BMOUtils.getICCConfig(ctx)
 	if err != nil {
 		log.WithError(err).Info("ICC configuration is not available")
 		return nil
 	}
 
-	var architectures []string
-	architectures, err = r.OcRelease.GetImageArchitecture(log, iccConfig.IronicAgentImage, infraEnvInternal.PullSecret)
+	log.Infof("Using ironic URLs from ICC config (Base: %s, Inspector: %s)", iccConfig.IronicBaseURL, iccConfig.IronicInspectorBaseUrl)
+	return iccConfig
+}
+
+func (r *PreprovisioningImageReconciler) getArchitectureCompatibleIronicAgentImageFromBMO(log logrus.FieldLogger, infraEnvInternal *common.InfraEnv, iccConfig *ICCConfig) string {
+	if iccConfig == nil || iccConfig.IronicAgentImage == "" {
+		log.Info("No ironic agent image in ICC config")
+		return ""
+	}
+
+	architectures, err := r.OcRelease.GetImageArchitecture(log, iccConfig.IronicAgentImage, infraEnvInternal.PullSecret)
 	if err == nil && funk.Contains(architectures, infraEnvInternal.CPUArchitecture) {
 		log.Infof("Setting ironic agent image (%s) from ICC config", iccConfig.IronicAgentImage)
-		return iccConfig
+		return iccConfig.IronicAgentImage
```

Comment:
Why pass the entire config if all we care about is the image? Can this function be something more like `func imageMatchesInfraenvArch(log, infraEnvInternal, ironicImage) bool`?